controller/src/main/java/org/apache/rocketmq/controller/impl/DLedgerController.java

```diff
@@ -193,6 +194,32 @@ public RemotingServer getRemotingServer() {
         return this.dLedgerServer.getRemotingServer();
     }
 
+    @Override
+    public CompletableFuture<RemotingCommand> cleanBrokerData(
+        final CleanControllerBrokerDataRequestHeader requestHeader,
+        final BiPredicate<String, String> brokerAlivePredicate) {
+
+        String brokerAddrs = requestHeader.getBrokerAddress();
+        String clusterName = requestHeader.getClusterName();
+        if ((null == brokerAddrs || brokerAddrs.trim().isEmpty()) && brokerAlivePredicate.test(clusterName, null)) {
+            RemotingCommand responseCommand = RemotingCommand.createResponseCommand(ResponseCode.CONTROLLER_INVALID_REQUEST,
+                String.format("Broker %s is still alive, clean up failure", requestHeader.getBrokerName()));
+            return CompletableFuture.completedFuture(responseCommand);
+        } else {
+            String[] brokerAddrArray = brokerAddrs.split(";");
+            for (String brokerAddr : brokerAddrArray) {
+                if (brokerAlivePredicate.test(clusterName, brokerAddr)) {
+                    RemotingCommand responseCommand = RemotingCommand.createResponseCommand(ResponseCode.CONTROLLER_INVALID_REQUEST,
+                        String.format("Broker [%s,  %s] is still alive, clean up failure", requestHeader.getBrokerName(), brokerAddr));
+                    return CompletableFuture.completedFuture(responseCommand);
+                }
+            }
+        }
```

Comment:
How about moving these to replicasInfoManager just like other methods do？

Reply:
I will moving these to replicasInfoManager